client/src/components/Questions/Questions.component.tsx

```diff
@@ -72,7 +72,7 @@ export const Questions = ({
           questionsPerPage || questionsDisplayState.questionsPerPage
         }
         showViewAllQuestionsButton={
-          showViewAllQuestionsButton ?? questionsDisplayState.value !== 'all'
+          showViewAllQuestionsButton ?? questionsDisplayState.value === 'top'
```

Comment:
Also, why did you change this?

Reply:
ah this was in response to a bug I found. `showViewAllQuestionsButton` is not a great name (but I was struggling to figure out what a better name is) because what it does is it either shows the "View all questions" button or it shows the pagination. 

After checking out the codebase, I noticed the behavior we want is to show the "View all questions" when viewing top questions. 

In the two other display states (i.e. currently viewing all questions OR viewing the questions of a selected topic), we really want to show the pagination. 

Hence this change!

Reply:
are you sure about the latter?

Reply:
I... think so? I guess to put it more succinctly, the "View all questions" button should only show up when:
- User is looking at the top questions, so the button brings them to all questions instead
- User is authenticated, in which case they'll be straight-up shown 50 questions, with no need to toggle to see more (I'm assuming the original design intent is 50 questions should be more than enough to see all questions...? I'm guessing this hard-coding might result in errors down the line...?)

In most cases, I don't think it makes a difference because users will typically access topic via the AgencyHomePage, but changing this ensures the following are consistent:

Local without this change:
<img width="946" alt="image" src="https://user-images.githubusercontent.com/67887489/157583594-98be68ce-647b-4f12-853a-7bac13069233.png">

Live AskGov:
<img width="938" alt="image" src="https://user-images.githubusercontent.com/67887489/157584086-1834a9d8-6c51-4338-acb9-e8db9fe1cb45.png">